src/barrier-picker/SpreadBarrierCard.js

```diff
@@ -42,7 +42,7 @@ export default class SpreadBarrierCard extends Component {
                 <InputGroup
                     type="number"
                     label={`Amount per point (${currency})`}
-                    value={spreadInfo.amountPerPoint}
+                    defaultValue={spreadInfo.amountPerPoint}
```

Comment:
It uses value for a reason, to preserve state when change mode, we need to use value